aws/resource_aws_dynamodb_table_item.go

```diff
@@ -224,18 +225,30 @@ func resourceAwsDynamoDbTableItemDelete(d *schema.ResourceData, meta interface{}
 
 func buildDynamoDbExpressionAttributeNames(attrs map[string]*dynamodb.AttributeValue) map[string]*string {
 	names := map[string]*string{}
+
 	for key := range attrs {
-		names["#a_"+key] = aws.String(key)
+		names["#a_"+cleanDynamoDbKeyName(key)] = aws.String(key)
 	}
 
+	log.Printf("[DEBUG] ExpressionAttributeNames: %+v", names)
 	return names
 }
 
+func cleanDynamoDbKeyName(key string) string {
+	reg, err := regexp.Compile("[^a-zA-Z]+")
```

Comment:
I couldn't find what is the exact constraint for the key name. Is it sort of "the key name must start with alpha char only" (rather than "alphanumeric")? If yes, could you please point me in the right direction in the AWS DynamoDB documentation? Thanks.